tests/kernel/smp/src/main.c

```diff
@@ -441,6 +444,127 @@ void test_wakeup_threads(void)
 	cleanup_resources();
 }
 
+/* a thread for testing SMP IPI */
+static void thread_ipi_entry(void *p1, void *p2, void *p3)
+{
+	int bsp_id = *(int *)p1;
+	int cpu_id = -1;
+
+	/* get current cpu number for running thread */
+	_cpu_t *curr_cpu = arch_curr_cpu();
+
+	/**TESTPOINT: call arch_curr_cpu() to get cpu struct */
+	zassert_true(curr_cpu != NULL,
+			"test failed to get current cpu.");
+
+	cpu_id = curr_cpu->id;
+
+	zassert_true(bsp_id != cpu_id,
+			"should not be the same with our BSP");
+
+	/* forever loop to wait for IPI*/
+	while (1) {
+	};
+}
+
+/* Define an our own IPI interrupt handler */
+__unused static void test_our_sched_ipi(void)
+{
+	sched_ipi_has_called++;
+}
+
+#if defined(CONFIG_X86_64)
+
+/* a pointer to IDT table */
+extern void (*x86_irq_funcs[])(void *);
+
+static void subsitute_original_ipi_handler(void)
+{
+	x86_irq_funcs[CONFIG_SCHED_IPI_VECTOR - IV_IRQS] =
+		(void *) test_our_sched_ipi;
+}
+
+static void change_back_ipi_handler(void)
+{
+	x86_irq_funcs[CONFIG_SCHED_IPI_VECTOR - IV_IRQS] =
+		(void *) z_sched_ipi;
+}
+#else
```

Comment:
You want something here that will cause a test failure, and a filter in the sanitycheck YAML for the platforms that are actually implemented, we don't want tests passing because they weren't implemented.

Reply:
OK, I already update this part. But I have a question, isn't other qemu platform support SMP except qemu_x86_64? By default sanitycheck only run this platform in this case.

Reply:
@enjiamai As I understand, you didn't implement in code idea suggested by Andy, right?

Reply:
@maksimmasalski No, not yet. I think I will try that first. It could be a more simply way to do that instead of implementing for different architecture. Thanks!